src/pages/index.js

```diff
@@ -65,34 +109,137 @@ profileEditBtn.addEventListener("click", () => {
+    api
+      .setProfilePicture(profilePictureFormInput.value)
+      .then((res) => {
+        profileAvatar.src = res.avatar;
+        profilePictureModal.resetButtonState();
+      })
+      .catch((err) => {
+        api.logError(err);
+      });
+    profilePictureModal.close();
+  },
+  "Save"
+);
+
+profilePictureModal.setEventListeners();
 
-const addModal = new PopupWithForm("#profile-add-modal", () => {
-  const { title, url } = addModal.getInputValues();
-  const newCard = renderCard({ name: title, link: url });
-  cardSection.addItem(newCard);
-  addModal.close();
+profilePictureEditBtn.addEventListener("click", () => {
+  profilePictureModal.open();
 });
 
+//--------------------ADD CARD POPUP-WITH-FORM INSTANTIATION-------------------->>
+
+const addModal = new PopupWithForm(
+  "#profile-add-modal",
+  () => {
+    addModal.loadingButtonState();
+    const { title, url } = addModal.getInputValues();
+    api
+      .addNewCard({ name: title, link: url })
+      .then((cardData) => {
+        const newCard = renderCard({
+          name: cardData.name,
+          link: cardData.link,
+          _id: cardData._id,
+          isLiked: cardData.isLiked,
+        });
+        cardSection.addItem(newCard);
+        addModal.resetButtonState();
+      })
+      .catch((err) => {
+        api.logError(err);
+      });
+    addModal.close();
+  },
+  "Create"
+);
+
 addModal.setEventListeners();
 
 addCardBtn.addEventListener("click", () => {
   addModal.open();
 });
 
+//--------------------DELETE CARD POPUP INSTANTIATION-------------------->>
+
+const deleteModal = new PopupWithConfirmation("#card-delete-modal");
+
+deleteModal.setEventListeners();
+
 //--------------------FUNCTIONS-------------------->>
 
 function renderCard(data) {
-  const cardElement = new Card(data, "#card-template", () => {
-    imagePopup.open(data);
-  });
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    () => {
+      imagePopup.open(data);
+    },
+    deleteModal,
+    () => {
+      const id = cardElement.getId();
+
+      deleteModal.open();
+
+      deleteModal.setSubmitAction(() => {
+        deleteModal.loadingButtonState();
+
+        api
+          .deleteCard(id)
+          .then(() => {
+            cardElement.deleteHandler();
+            deleteModal.close();
+            deleteModal.resetButtonState();
+          })
+          .catch((err) => {
+            api.logError(err);
+          });
+      });
+    },
+    () => {
+      const id = cardElement.getId();
+
+      if (!data.isLiked) {
+        api
+          .addCardLike(id)
+          .then(() => {
+            cardElement.likeHandler();
+          })
+          .catch((err) => {
+            api.logError(err);
+          });
+      } else {
+        api
+          .removeCardLike(id)
+          .then(() => {
+            cardElement.likeHandler();
+          })
+          .catch((err) => {
+            api.logError(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors